include/api/CDataFrameAnalysisRunner.h

```diff
@@ -192,19 +208,41 @@ class API_EXPORT CDataFrameAnalysisRunner {
 class API_EXPORT CDataFrameAnalysisRunnerFactory {
 public:
     using TRunnerUPtr = std::unique_ptr<CDataFrameAnalysisRunner>;
+    using TDataFrameUPtrTemporaryDirectoryPtrPr =
+        CDataFrameAnalysisRunner::TDataFrameUPtrTemporaryDirectoryPtrPr;
 
 public:
     virtual ~CDataFrameAnalysisRunnerFactory() = default;
     virtual const std::string& name() const = 0;
 
-    TRunnerUPtr make(const CDataFrameAnalysisSpecification& spec) const;
+    //! Create a new runner object from \p spec.
+    //!
+    //! \param[in] spec The analysis specification.
+    //! \param[out] frameAndDirectory If non-null a data frame is created which
+    //! is suitable for the analysis together with the directory handle, if it
+    //! is stored on disk, and written to this.
+    TRunnerUPtr make(const CDataFrameAnalysisSpecification& spec,
+                     TDataFrameUPtrTemporaryDirectoryPtrPr* frameAndDirectory = nullptr) const;
```

Comment:
A significant part of this change (particular in the test code) relates to handing off the creation of the data frame on which to run the analysis to this class. This is needed because we can restore data if we are running incremental training. This is optional because we don't want to restore or create a data frame if we are just estimating memory.